### PR TITLE
add tool-tip to remove icon

### PR DESCRIPTION
**Pull request by debsmita1**
**Jira Stroy:**
https://issues.redhat.com/browse/ODC-4247

**Solution Description:**
This PR adds "Remove" tooltip to `MinusCircleIcon` & "Add" tooltip to `PlusCircleIcon` button

**Screenshots:**
Screenshot from 2020-07-15 13-49-09
Screenshot from 2020-07-15 13-33-34
Screenshot from 2020-07-15 13-33-27
Screenshot from 2020-07-15 13-30-57
Screenshot from 2020-07-15 13-30-47
Screenshot from 2020-07-15 13-30-40
Screenshot from 2020-07-15 13-58-43
Screenshot from 2020-07-15 13-57-59
Screenshot from 2020-07-15 15-17-54
Screenshot from 2020-07-15 15-58-45
Screenshot from 2020-07-15 22-48-53

### CODE REVIEW

**Comment by debsmita1** [comment]
/cc @beaumorley @serenamarie125 ^
Here https://issues.redhat.com/browse/ODC-4248 I have added a list which contains the areas that I have fixed

File: frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/PipelineParameters.tsx [hunk 1]
```diff
@@ -9,13 +9,12 @@ type PipelineParametersProps = {
...
 
 const PipelineParameters: React.FC<PipelineParametersProps> = (props) => {
-  const { addLabel = 'Add Pipeline Parameter', fieldName, isReadOnly = false } = props;
+  const { fieldName, isReadOnly = false } = props;
   const emptyMessage = 'No parameters are associated with this pipeline.';
 
   return (
     <MultiColumnField
       name={fieldName}
-      addLabel={addLabel}
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/PipelineParameters.tsx:18]: Hmm, was this needed to be removed?

File: frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/PipelineResources.tsx [hunk 2]
```diff
@@ -10,12 +10,11 @@ type PipelineResourcesParam = {
 };
 
 const PipelineResources: React.FC<PipelineResourcesParam> = (props) => {
-  const { addLabel = 'Add Pipeline Resource', fieldName, isReadOnly = false } = props;
+  const { fieldName, isReadOnly = false } = props;
   const emptyMessage = 'No resources are associated with this pipeline.';
   return (
     <MultiColumnField
       name={fieldName}
-      addLabel={addLabel}
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/PipelineResources.tsx:18]: Same here; was this needed?

File: frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx [hunk 3]
```diff
@@ -23,10 +23,8 @@ const ProjectAccessForm: React.FC<FormikProps<FormikValues>> = ({
     <div className="co-m-pane__form">
       <MultiColumnField
         name="projectAccess"
-        addLabel="Add Access"
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx:26]: Not sure we want to drop this... it looks odd imo:
image

**Comment by andrewballantyne** [comment]
Yeah, not a fan of this:
image

I think in this case the `Add` should have more description as there are two types of "add"s for this item.

**Comment by debsmita1** [comment]
@andrewballantyne - @beaumorley & I conversed on whether to have simply `Add` or `Add <context>`. The conversation is here https://issues.redhat.com/browse/ODC-4248. I have added a list of all the places I have made the changes in the mentioned subtask. We can collaborate and decide on which areas we want to have the context mentioned with Add/Remove & which areas we want it simply Add/Remove.

Add/Remove can be the default addString/tooltip & if user wants a customized one it can be passed as props.

File: frontend/packages/dev-console/src/components/health-checks/RequestTypeForms.tsx [hunk 4]
```diff
@@ -71,7 +71,6 @@ export const HTTPRequestTypeForm: React.FC<RequestTypeFormProps> = ({ probeType
           nameValuePairs={nameValue}
           valueString="Value"
           nameString="Header Name"
-          addString="Add Header"
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/health-checks/RequestTypeForms.tsx:74]: I referred to this doc https://docs.google.com/document/d/1i_mPwL5ZNwrkSHIYAorQjIZDJDLvD6ikffNkq52eTFA/edit?ts=5e94bacb (page 3) while working on the health checks feature

**Comment by andrewballantyne** [comment]
> @andrewballantyne - @beaumorley & I conversed on whether to have simply `Add` or `Add <context>`. The conversation is here https://issues.redhat.com/browse/ODC-4248. I have added a list of all the places I have made the changes in the mentioned subtask. We can collaborate and decide on which areas we want to have the context mentioned with Add/Remove & which areas we want it simply Add/Remove.
> 
> Add/Remove can be the default addString/tooltip & if user wants a customized one it can be passed as props.

I think on paper it looked good... imo it does not look good in practice. Remove has context of what it is... but Add does not always have context. I think we should leave "Add" alone and not truncate it down. 

If there are no rows, we are making a big leap to the user reading the block of text that is above it; my experience in what users do / do not do... is read text lol. I think the Add "button" being more descriptive is better imo.

@beaumorley Definitely will need you to weigh in here.

**Comment by beaumorley** [comment]
@andrewballantyne I think you are saying that sometimes "Add" alone is not clear enough. If there is nothing there and they see "Add" they may not know what they are adding.  I like to keep the verbiage down when we can but I see what you are saying. If we want to add specific Add <xxx> (which sounds like a good call) we would like to help create the copy. I think we could do this after HLD? Could we use the list that was generated in the subtask to do that or do you have a lookup table somewhere?

I would say we should offer this option to include context to both Add and Remove. Then we can decide if it is needed.

Thanks! @debsmita1

**Comment by andrewballantyne** [comment]
> @andrewballantyne I think you are saying that sometimes "Add" alone is not clear enough. If there is nothing there and they see "Add" they may not know what they are adding. I like to keep the verbiage down when we can but I see what you are saying. If we want to add specific Add (which sounds like a good call) we would like to help create the copy. I think we could do this after HLD? Could we use the list that was generated in the subtask to do that or do you have a lookup table somewhere?
> 
> I would say we should offer this option to include context to both Add and Remove. Then we can decide if it is needed.
> 
> Thanks! @debsmita1

Well, it's definitely your call @beaumorley. My preference is the Story was to focus on the tooltip for Remove and we expanded it to truncate the "Add", which imo is not a good idea.

The PR looks fine from a technical point of view, I just don't believe we needed to touch the "Add" _context_ text that helped give the add button definitive meaning. 

/lgtm

We can address it after the fact for sure as one-off bugs or whatever as the real issues arise.

**Comment by andrewballantyne** [comment]
/assign @christianvogt 

Need your approval here.

File: frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx (hunk 3 above)
Comment (christianvogt) [reply frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx:26]: @beaumorley the above screen from Andrew looks odd.

In other places we follow this pattern where there is secondary text saying there are no _list items_:
image


If we want to follow the same, we should log an issue for it.

**Pull request merged**